gprofiler/profilers/perf.py

```diff
@@ -322,27 +323,26 @@ def wait_and_script(self) -> str:
             # (unlike Popen.communicate())
             assert self._process is not None and self._process.stderr is not None
             logger.debug(f"{self._log_name} run output", perf_stderr=self._process.stderr.read1())  # type: ignore
-
         try:
-            inject_data = Path(f"{str(perf_data)}.inject")
-            if self._inject_jit:
-                run_process(
-                    [perf_path(), "inject", "--jit", "-o", str(inject_data), "-i", str(perf_data)],
+            with NamedTemporaryFile(dir=os.path.dirname(self._output_path), suffix=".inject") as inject_data:
+                if self._inject_jit:
+                    run_process(
+                        [perf_path(), "inject", "--jit", "-o", str(inject_data), "-i", str(perf_data)],
+                    )
+                    perf_data = Path(inject_data.name)
```

Comment:
This creates a problematic flow:
1. `perf_data = temporary file`
2. Exception is raised during `perf script`
3. `finally` runs and tries to unlink `perf_data` which is already unlinked.

I suggest you use a 3rd variable (like `perf_script_input`), then set it to either `perf_data` OR `inject_data`, and only unlink `perf_data` as `inject_data` is automatically removed.